python/pyspark/sql/tests/test_arrow.py

```diff
@@ -383,6 +383,16 @@ def test_timestamp_dst(self):
         assert_frame_equal(pdf, df_from_python.toPandas())
         assert_frame_equal(pdf, df_from_pandas.toPandas())
 
+    # Regression test for SPARK-28003
+    def test_timestamp_nat(self):
+        dt = [pd.NaT, pd.Timestamp('2019-06-11'), None] * 100
+        pdf = pd.DataFrame({'time': dt})
+
+        for arrow_enabled in [False, True]:
```

Comment:
you can just use `_toPandas_arrow_toggle` which returns 2 DataFrames with and without arrow

Reply:
Ah thanks for the tip! (I ended up using _createDataFrame_toggle because that's the path I want to test)